packages/cli/src/lib/cmds/play.ts

```diff
@@ -50,62 +50,52 @@ class Play {
     // Validate that the publish value is listed in the available Tracks.
     // If no value was supplied with publish we make it internal.
     const userSelectedTrack = asPlayStoreTrack(this.args.publish?.toLowerCase() || 'internal');
     if (userSelectedTrack == null) {
       this.prompt.printMessage(enUS.messageInvalidTrack);
       return false;
     }
-    // appbundlelocation is an option argument.
-    if (this.args.appBundleLocation && fs.existsSync(this.args.appBundleLocation!!)) {
-      await this.googlePlay.publishBundleWithGradle(userSelectedTrack, this.args.appBundleLocation);
-      return true;
+
+    // Where we should find our default output file
+    const defaultPath = path.join(process.cwd(), defaultSignedAppBundleFileName);
+
+    const publishFilePath = this.args.appBundleLocation || defaultPath;
+
+    if (!fs.existsSync(publishFilePath)) {
+      throw new Error(`App Bundle not found on disk at location: ${publishFilePath}`);
     }
-    // Make tmp directory copy file over signed APK then cleanup.
-    const publishDir = fs.mkdtempSync('bubblewrap');
-    const signedAppBundleFileName = 'app-release-bundle.aab';
-    // Where we should find our output file
-    const defaultPath = path.join(process.cwd(), signedAppBundleFileName);
 
-    fs.copyFileSync(defaultPath, path.join(publishDir, signedAppBundleFileName));
-    await this.googlePlay.publishBundleWithGradle(userSelectedTrack, publishDir);
+    // TODO(@nohe427): Add cli commands for retained bundles.
+    const retainedBundles = [] as number[];
+    await this.googlePlay.publishBundle(
+        userSelectedTrack, publishFilePath, twaManifest.packageId, retainedBundles);
 
-    fs.unlinkSync(path.join(publishDir, signedAppBundleFileName));
-    fs.rmdirSync(publishDir);
     return true;
   }
 
   /**
-  * Validates that the service account JSON file exists.
-  * @param {string | undefined} path - The path the the JSON file.
-  * @return {boolean} Whether or not the JSON file exists.
+  * Updates the gradle file based on the updates to the twa-manifest.json file and warns the user
+  *   that they need to call build again.
+  * @param {string} manifestFile - The path to the the TwaManifest JSON file.
+  * @param {string} appVersionName - Optional: Changes the string representation of the version.
   */
-  private validServiceAccountJsonFile(path: string | undefined): boolean {
-    if (path == undefined) {
-      // Log an error
-      return false;
-    }
-    if (!fs.existsSync(path)) {
-      // path doesn't exist log an error
-      return false;
-    }
-    return true;
-  }
-
-  private async updateProjectAndWarn(manifestFile: string): Promise<void> {
+  private async updateProjectAndWarn(manifestFile: string, appVersionName?: string): Promise<void> {
     await updateProject(
-        true, null, this.prompt, this.args.targetDirectory || process.cwd(), manifestFile);
+        true,
+        appVersionName || null,
```

Comment:
Do we need `appVersionName || null`? `appVersionName` is either going to have a value or be `null` anyway. We can just pass `appVersionName`.

Reply:
Yes because appVersionName in updateProject takes `string | null`. appVersionName is `string | undefined`. Would it be better to update project here to take `string | undefined`?